sympy/physics/mechanics/_system.py

```diff
@@ -0,0 +1,937 @@
+from sympy.utilities.iterables import iterable
+from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
+from sympy.physics.mechanics.functions import _validate_coordinates, Lagrangian
+from sympy.physics.mechanics.body_base import BodyBase
+from sympy.physics.mechanics.particle import Particle
+from sympy.physics.mechanics.loads import Force, Torque, _parse_load, gravity
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.mechanics.method import _Methods
+from sympy.physics.mechanics.kane import KanesMethod
+from sympy.physics.mechanics.lagrange import LagrangesMethod
+from functools import wraps
+
+__all__ = ['System']
+
+
+def _reset_eom_method(method):
+    """Decorator to reset the eom_method if a property is changed."""
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        self._eom_method = None
+        return method(self, *args, **kwargs)
+
+    return wrapper
+
+
+class System(_Methods):
+    """Class to define a system and form the equations of motion.
+
+    Explanation
+    ===========
+
+    A ``System`` instance stores the different objects associated with a model,
+    including bodies, joints, constraints, and other relevant information. With
+    all the relationships between components defined, the ``System`` can be used
+    to form the equations of motion using a backend, such as ``KanesMethod``.
+    The ``System`` has been designed to be compatible with third-party
+    libraries for greater flexibility and integration with other tools.
+
+    Attributes
+    ==========
+
+    origin : Point
+        Global origin of the system.
+    frame : ReferenceFrame
+        Inertial reference frame of the system.
+    x : Vector
+        Unit vector in the x direction of the inertial reference frame.
+    y : Vector
+        Unit vector in the y direction of the inertial reference frame.
+    z : Vector
+        Unit vector in the z direction of the inertial reference frame.
+    q : Matrix
+        Matrix of all the generalized coordinates.
+    u : Matrix
+        Matrix of all the generalized speeds.
+    q_ind : Matrix
+        Matrix of the independent generalized coordinates.
+    q_dep : Matrix
+        Matrix of the dependent generalized coordinates.
+    u_ind : Matrix
+        Matrix of the independent generalized speeds.
+    u_dep : Matrix
+        Matrix of the dependent generalized speeds.
+    kdes : Matrix
+        Matrix of the kinematic differential equations.
+    bodies : tuple of BodyBase subclasses
+        Tuple of all bodies that make up the system.
+    joints : tuple of Joint
+        Tuple of all joints that connect bodies in the system.
+    loads : tuple of LoadBase subclasses
+        Tuple of all loads that have been applied to the system.
+    holonomic_constraints : Matrix
+        Matrix with the holonomic constraints as rows.
+    nonholonomic_constraints : Matrix
+        Matrix with the nonholonomic constraints as rows.
+    eom_method : class
```

Comment:
We don't constrain this at all, like specifying that it needs to be a subclass of `_Methods` so that it at least somewhat defines the correct interface?

Reply:
Will change it to `subclass of KanesMethod or LagrangesMethod`.

Reply:
Addressed in 9d9a8c1cc3532150f0e4764d9c5a6016cc80a939